### ISSUE TITLE
[Feature request] Add custom command on action

### ISSUE THREAD

**Issue by bonswouar** [Schaka/janitorr#95]
It could be quite useful to be able to configure a custom command (or a custom call to an URL?) when an action is done, i.e when a movie is deleted, or added/removed from Leaving Soon _(but not sure the second part is possible as I don't think Janitorr keeps track of diffs?)_

My use case would be to call a webhook (with the file/directory as parameter) on [Autopulse](https://github.com/dan-online/autopulse/), to have an efficient synchronization of Jellyfin's library (I believe because of symlinks it doesn't work that well for Leaving Soon on my instance by default)

**Comment by bonswouar** [comment]
Also for more use cases, it could also be triggered for failed actions, end of main loop, etc, with some Type as a param to differentiate them.
(I was thinking for example to trigger a notification when "Can't delete movie" happens)

**Comment by Schaka** [comment]
![image](https://github.com/user-attachments/assets/84587291-86b8-4c57-a9d8-7aafdcd1e9a3)

Would this be different from Radarr/Sonarr Webhooks?
I don't think it's in the scope of Janitorr to provide something like it.

It's definitely technically possible (on delete or failure to delete), but it's too close to existing functionality in the arrs and I'm trying to keep dependencies to a minimum. I was already pretty wary of adding Bazarr. 

You're right that there's no info regarding the Leaving Soon collection, except when adding something.

**Comment by bonswouar** [comment]
@Schaka For sake of simplicity and dependency-free, I was thinking just a command, then the user can do whatever he wants with it!
That's what Bazarr has done:
![image](https://github.com/user-attachments/assets/6a147e1a-4f79-4812-9bec-041d41620399)

Imo it should cover most useful cases, and could be quite easy to implement. What do you think?

**Comment by Schaka** [comment]
You're correct. The container doesn't have access to random shell commands. It might be something that's possible to add, but if I were to add anything, I'd likely make it a generalized POST.

I'll keep this ticket open, but don't hold your breath. Janitorr is roughly where I want it to be and I've moved most of my development efforts over to Naviseerr.

**Comment by Schaka** [comment]
Feel free to submit a PR that would solve this. You would likely:

- need to find a way to install custom dependencies into the Buildpack provided image
- or build and supply your own base image, compatible with x86 and aarch64 for buildpack to use instead
- create a new map of CustomActionType (enum) => "custom action" property under `JellyfinProperties`
- call that custom action on of type x whenever an a certain use case is hit (like deleted media, failed to delete media, added leaving soon)

**Comment by bonswouar** [comment]
@Schaka A webhook would be perfect!
FYI for Janitorr's use case, no payload content is needed, only the URL is used, as per the doc:
```bash
$ curl -u 'admin:password' 'http://localhost:2875/triggers/manual?path=/path/to/file&hash=1234567890' 
# or
$ curl -H 'Authorization: Basic <base_64_encoded_login>' 'http://localhost:2875/triggers/manual?path=/path/to/file&hash=1234567890'
```
But for a more general approach, a webhook with a list of parameters that can be used in the URL, in a JSON payload, and as headers, would probably cover most needs.
I would certainly use it to add [ntfy](https://docs.ntfy.sh/publish/) notifications also :)

**Comment by hacshacdgacs** [comment]
Just jumping on this, I was never able to get the "leaving-soon" notifications working through Jellyfin so having janitorr provide a webhook feature of it's own would be appreciated
